com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/AtlasUtil.java

```diff
@@ -47,33 +47,50 @@
 
 public class AtlasUtil {
     public static class MappedAnimDesc extends AnimDesc {
-        List<String> ids;
+        private List<String> paths;
+        private List<String> ids;
+        private boolean singleFrame;
 
-        public MappedAnimDesc(String id, List<String> ids, Playback playback, int fps, boolean flipHorizontal,
-                boolean flipVertical) {
+        public MappedAnimDesc(String id, List<String> paths, List<String> ids, Playback playback, int fps, boolean flipHorizontal, boolean flipVertical) {
             super(id, playback, fps, flipHorizontal, flipVertical);
+            this.paths = paths;
             this.ids = ids;
+            this.singleFrame = false;
         }
 
-        public MappedAnimDesc(String id, List<String> ids) {
+        public MappedAnimDesc(String id, List<String> paths, List<String> ids) {
             super(id, Playback.PLAYBACK_NONE, 0, false, false);
+            this.paths = paths;
             this.ids = ids;
+            this.singleFrame = true; // This animation is from a single frame animation
         }
 
-        public List<String> getIds() {
+        public List<String> getIds() { // For testing
             return this.ids;
         }
+
+        public List<String> getPaths() {
+            return this.paths;
+        }
+
+        @Override
+        public String getFrameId(int frameIndex) { // local frame index
```

Comment:
Opted for an override for simplicity.
In general, I think the animation loading/iteration could be much cleaner, with more specific methods/attributes, and an explicit struct for the animation frames (index, path, id)